xclim/core/datachecks.py

```diff
@@ -59,3 +65,42 @@ def check_daily(var: xr.DataArray):
     This does not check for gaps in series.
     """
     return check_freq(var, "D")
+
+
+@datacheck
+def check_common_time(*inputs):
+    """Raise an error if the list of inputs doesn't have a single common frequency.
+
+    Raises
+    ------
+    ValidationError
+        - if the frequency of any input can't be inferred
+        - if inputs have differing frequencies
+        - if inputs have a daily or hourly frequency, but they are not given at the same time of day.
+
+    Parameters
+    ----------
+    inputs : xr.DataArray
+        Input arrays.
+    """
+    # Check all have the same freq
+    freqs = {xr.infer_freq(da.time) for da in inputs}
```

Comment:
Suggest using a list comprehension, then convert to a set to check for unicity. This way, you can print the list in the error message.